dbms/src/Storages/DeltaMerge/Segment.cpp

```diff
@@ -170,16 +177,32 @@ StableValueSpacePtr createNewStable( //
     PageIdU64 stable_id,
     WriteBatches & wbs)
 {
-    auto delegator = context.path_pool.getStableDiskDelegator();
+    auto delegator = context.path_pool->getStableDiskDelegator();
     auto store_path = delegator.choosePath();
 
-    PageIdU64 dtfile_id = context.storage_pool.newDataPageIdForDTFile(delegator, __PRETTY_FUNCTION__);
+    PageIdU64 dtfile_id = context.storage_pool->newDataPageIdForDTFile(delegator, __PRETTY_FUNCTION__);
     auto dtfile = writeIntoNewDMFile(context, schema_snap, input_stream, dtfile_id, store_path);
 
     auto stable = std::make_shared<StableValueSpace>(stable_id);
     stable->setFiles({dtfile}, RowKeyRange::newAll(context.is_common_handle, context.rowkey_column_size));
     stable->saveMeta(wbs.meta);
-    wbs.data.putExternal(dtfile_id, 0);
+    if (auto data_store = context.db_context.getRemoteDataStore(); !data_store)
+    {
+        wbs.data.putExternal(dtfile_id, 0);
+    }
+    else
+    {
+        auto store_id = context.db_context.getTMTContext().getKVStore()->getStoreID();
+        // FIXME: get table_id from context
+        Remote::DMFileOID oid{.store_id = store_id, .table_id = 0, .file_id = dtfile_id};
```

Comment:
We already have physical_table_id in the DMContext?

Reply:
added in the latest commit